### PR TITLE
Implement S3 file upload API route

### PR DESCRIPTION
**Pull request by stefanpython**
This PR adds a new API route for handling file uploads to AWS S3.

Key changes:
- Implement `uploadFileToS3` function to manage S3 uploads
- Add POST route for file validation and upload
- Implement file type checking for images
- Ensure correct content type is passed to S3

Testing:
- [x] Tested file uploads with various image types
- [x] Verified S3 upload functionality
- [x] Checked error handling for invalid file types

Please review and let me know if any changes are needed.

### CODE REVIEW

File: frontend/app/api/s3-upload/route.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { NextResponse } from "next/server";
```
Comment (dIB59) [inline frontend/app/api/s3-upload/route.ts:1]: Make this file server only, to avoid unintended client usage of server code

**Review by dIB59 — CHANGES_REQUESTED** [review]
Take a look at the docs
https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment

File: frontend/app/api/s3-upload/route.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
+
+/**
+ * Initialize S3 client with credentials from environment variables
+ */
+const s3Client = new S3Client({
+    region: process.env.NEXT_PUBLIC_AWS_S3_REGION as string,
+    credentials: {
+        accessKeyId: process.env.NEXT_PUBLIC_AWS_S3_ACCESS_KEY_ID as string,
```
[the remark this answers is not in the document]
Comment (stefanpython) [reply frontend/app/api/s3-upload/route.ts]: Made variables private and installed server-only packaged for extra security layer

**Comment by stefanpython** [comment]
> Take a look at the docs https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment

Thank you for the support and for providing me with the docs. I appreciate it.

**Review by dIB59 — APPROVED** [review]
LGTM

**Pull request merged**